sharing_nodes.go

```diff
@@ -0,0 +1,37 @@
+package headscale
+
+import "gorm.io/gorm"
+
+const errorSameNamespace = Error("Destination namespace same as origin")
+const errorNodeAlreadyShared = Error("Node already shared to this namespace")
+
+// Sharing is a join table to support sharing nodes between namespaces
+type SharedNode struct {
+	gorm.Model
+	MachineID   uint64
+	Machine     Machine
+	NamespaceID uint
+	Namespace   Namespace
+}
+
+// ShareNodeInNamespace adds a machine as a shared node to a namespace
+func (h *Headscale) ShareNodeInNamespace(m *Machine, ns *Namespace) error {
```

Comment:
The wording of this is a bit confusing, it sounds like we are checking if it is in a namespace, e.g.
```go
if ShareNodeInNamespace() {}
```

How about:
`AddSharedNodeToNamespace`

Also we are getting quite inconsistent with `Machine` vs `Node`, Should this be `SharedMachine` ?

Reply:
Changed to `SharedMachine`. It makes sense, having `Machine` as the main DB object.